samples/bluetooth/direct_test_mode/src/dtm.c

```diff
@@ -0,0 +1,1351 @@
+	0x00, 0x21, 0x86, 0x9C, 0x6A, 0xD8, 0xCB, 0x4E,
+	0x14, 0x6A, 0xF9, 0x4D, 0xD2, 0x7E, 0xB2, 0x32,
+	0x03, 0xC6, 0x14, 0x4B, 0x7F, 0xD1, 0xB8, 0xA6,
+	0x79, 0x7C, 0x17, 0xAC, 0xED, 0x06, 0xAD, 0xAF,
+	0x0A, 0x94, 0x7A, 0xBA, 0x03, 0xE7, 0x92, 0xD7,
+	0x15, 0x09, 0x73, 0xE8, 0x6D, 0x16, 0xEE, 0xE1,
+	0x3F, 0x78, 0x1F, 0x9D, 0x09, 0x52, 0x6E, 0xF1,
+	0x7C, 0x36, 0x2A, 0x71, 0x6C, 0x75, 0x64, 0x44,
+	0x80, 0x10, 0x43, 0x4E, 0x35, 0xEC, 0x65, 0x27,
+	0x0A, 0xB5, 0xFC, 0x26, 0x69, 0x3F, 0x59, 0x99,
+	0x01, 0x63, 0x8A, 0xA5, 0xBF, 0x68, 0x5C, 0xD3,
+	0x3C, 0xBE, 0x0B, 0xD6, 0x76, 0x83, 0xD6, 0x57,
+	0x05, 0x4A, 0x3D, 0xDD, 0x81, 0x73, 0xC9, 0xEB,
+	0x8A, 0x84, 0x39, 0xF4, 0x36, 0x0B, 0xF7
+};
+
+static int clock_init(void)
+{
+	int err;
+	struct device *clock;
+	enum clock_control_status clock_status;
+
+	clock = device_get_binding(DT_INST_LABEL(0));
+	if (!clock) {
+		printk("Unable to find clock device binding\n");
+		return -ENODEV;
+	}
+
+	err = clock_control_on(clock, CLOCK_CONTROL_NRF_SUBSYS_HF);
+	if (err) {
+		printk("Unable to turn on the clock: %d", err);
+		return err;
+	}
+
+	do {
+		clock_status = clock_control_get_status(clock,
+			CLOCK_CONTROL_NRF_SUBSYS_HF);
+	} while (clock_status != CLOCK_CONTROL_STATUS_ON);
+
+	return 0;
+}
+
+static void timer_handler(nrf_timer_event_t event_type, void *context)
+{
+	/* Do nothing, use timer in polling mode. */
+}
+
+static int timer_init(void)
+{
+	nrfx_err_t          err;
+	nrfx_timer_config_t timer_cfg = {
+		.frequency = NRF_TIMER_FREQ_1MHz,
+		.mode      = NRF_TIMER_MODE_TIMER,
+		.bit_width = NRF_TIMER_BIT_WIDTH_16,
+	};
+
+	err = nrfx_timer_init(&dtm_inst.timer, &timer_cfg, timer_handler);
+	if (err != NRFX_SUCCESS) {
+		printk("nrfx_timer_init failed with: %d\n", err);
+		return -EAGAIN;
+	}
+
+	IRQ_CONNECT(TIMER0_IRQn, 6,
+		nrfx_timer_0_irq_handler, NULL, 0);
+
+	/* Timer is polled, but enable the compare0 interrupt in order to
+	 * wakeup from CPU sleep.
+	 */
+	nrfx_timer_extended_compare(&dtm_inst.timer,
+		NRF_TIMER_CC_CHANNEL0,
+		nrfx_timer_us_to_ticks(&dtm_inst.timer, TX_INTERVAL),
+		NRF_TIMER_SHORT_COMPARE0_CLEAR_MASK,
+		false);
+
+	nrfx_timer_compare(&dtm_inst.timer,
+		NRF_TIMER_CC_CHANNEL1,
+		nrfx_timer_us_to_ticks(&dtm_inst.timer, DTM_UART_POLL_CYCLE),
+		false);
+
+	dtm_inst.current_time = 0;
+	nrfx_timer_enable(&dtm_inst.timer);
+
+	return 0;
+}
+
+static int anomaly_timer_init(void)
+{
+	nrfx_err_t          err;
+	nrfx_timer_config_t timer_cfg = {
+		.frequency = NRF_TIMER_FREQ_125kHz,
+		.mode      = NRF_TIMER_MODE_TIMER,
+		.bit_width = NRF_TIMER_BIT_WIDTH_16,
+	};
+
+	err = nrfx_timer_init(&dtm_inst.anomaly_timer, &timer_cfg,
+			      timer_handler);
+	if (err != NRFX_SUCCESS) {
+		printk("nrfx_timer_init failed with: %d\n", err);
+		return -EAGAIN;
+	}
+
+	nrfx_timer_compare(&dtm_inst.timer,
+		NRF_TIMER_CC_CHANNEL0,
+		nrfx_timer_ms_to_ticks(&dtm_inst.timer,
+				       BLOCKER_FIX_WAIT_DEFAULT),
+		false);
+
+	return 0;
```

Comment:
Where is the IRQ of this timer enabled in NVIC? Or is the timer used in a blocking way only?